openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pages/Webhooks/WebhooksPageTest.java

```diff
@@ -0,0 +1,173 @@
+      Events.click(webDriver, webhooks.getEntityCreatedMenu());
+      Events.click(webDriver, webhooks.allEntities());
+      actions.click();
+      actions.perform();
+      Events.click(webDriver, webhooks.getSaveWebhook());
+    }
+    WebElement errorMessage = webDriver.findElement(webhooks.getToast());
+    if (errorMessage.isDisplayed()) {
+      Assert.assertEquals(errorMessage.getText(), "Request failed with status code 409");
+    }
+  }
+
+  @Test
+  void checkBlankName() throws InterruptedException {
+    webDriver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+    openWebHookPage();
+    Events.click(webDriver, webhooks.addWebhook());
+    Events.sendKeys(webDriver, webhooks.name(), "");
+    Events.click(webDriver, webhooks.getDescriptionBox());
+    Events.sendKeys(webDriver, webhooks.getDescriptionBox(), "test");
+    Events.sendKeys(webDriver, webhooks.getEndpoint(), "test.com");
+    Events.click(webDriver, webhooks.checkbox());
+    Thread.sleep(waitTime);
+    Events.click(webDriver, webhooks.getEntityCreatedMenu());
+    Events.click(webDriver, webhooks.allEntities());
+    actions.click();
+    actions.perform();
+    Events.click(webDriver, webhooks.getSaveWebhook());
+    WebElement errorMessage = webDriver.findElement(common.errorMessage());
+    if (errorMessage.isDisplayed()) {
+      Assert.assertEquals(errorMessage.getText(), "Webhook name is required.");
+    }
+  }
+
+  @Test
+  void checkBlankEndpoint() {
+    webDriver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+    openWebHookPage();
+    Events.click(webDriver, webhooks.addWebhook());
+    Events.sendKeys(webDriver, webhooks.name(), "test");
+    Events.click(webDriver, webhooks.getDescriptionBox());
+    Events.sendKeys(webDriver, webhooks.getDescriptionBox(), "test");
+    Events.sendKeys(webDriver, webhooks.getEndpoint(), "");
+    Events.click(webDriver, webhooks.checkbox());
+    Events.click(webDriver, webhooks.getEntityCreatedMenu());
+    Events.click(webDriver, webhooks.allEntities());
+    actions.click();
+    actions.perform();
+    Events.click(webDriver, webhooks.getSaveWebhook());
+    WebElement errorMessage = webDriver.findElement(common.errorMessage());
+    if (errorMessage.isDisplayed()) {
+      Assert.assertEquals(errorMessage.getText(), "Webhook endpoint is required.");
+    }
+  }
+
+  @Test
+  void checkBlankEntityCheckbox() {
+    webDriver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));
+    openWebHookPage();
+    Events.click(webDriver, webhooks.addWebhook());
+    Events.sendKeys(webDriver, webhooks.name(), "test");
+    Events.click(webDriver, webhooks.getDescriptionBox());
+    Events.sendKeys(webDriver, webhooks.getDescriptionBox(), "test");
+    Events.sendKeys(webDriver, webhooks.getEndpoint(), "test.com");
+    Events.click(webDriver, webhooks.getSaveWebhook());
+    WebElement errorMessage = webDriver.findElement(common.errorMessage());
+    if (errorMessage.isDisplayed()) {
```

Comment:
Should the test fail if error message is not displayed?

Reply:
The test is to check if error message is displayed on sending blank endpoint or blank name. Should I surround the error message with try catch block and throw exception if no such element exists?

Reply:
We expect errorMessage to be displayed. 

So the test would be:
```
if(errorMessae.IsDisplayed()) {
  Assert.assertEquals(errorMessage.getText(), "Webhook event filters are required.");
} else {
  Assert.fail("Expected error message to be displayed")
}
```

Reply:
We can simplify this further...

```
Assert.true(errorMessae.IsDisplayed());
Assert.assertEquals(errorMessage.getText(), "Webhook event filters are required.");
```

Reply:
Okay,I will make the changes

Reply:
@kshinde2512  We should avoid conditions in E2E tests when we are expecting some particular behaviour to occur. This applies for positive as well as negative test cases.

PS: @mithmatt correct me if I have misjudged.